src/app/register/register.component.ts

```diff
@@ -247,8 +242,42 @@ export class RegisterComponent implements OnInit, OnDestroy, AfterViewInit {
     return !!this.verificationCodeSupplied || this.emailVerificationToken?.valid || ! flags.requireEmailVerification
   }
 
-  registerPostDonation() {
-    alert(
-      "Function still to be built to register using donor UUID and Password and secret token, UUID: " + this.emailVerificationToken?.person_uuid + " secret token: " + this.emailVerificationToken?.secretNumber)
+  async registerPostDonation() {
+    this.processing = true;
+    this.registerPostDonationForm.markAllAsTouched();
+    const password = this.registerPostDonationForm.value.password
+
+    const passwordErrors = this.registerPostDonationForm.controls.password.errors;
+    if (passwordErrors?.required) {
+      this.error = "Password is required"
+      this.processing = false
+      return;
+    }
```

Comment:
really we should be checking for other possible errors as well here, e.g. the new password being too short. But if we don't check it it's not that bad, identity server checks it the donor sees a readable error message.

I feel like there must be a much better way to handle form errors, it should be too hard to iterate through all errors on a form or form control and make a human readable message.  But that seems like something that should be in a library for Angular rather than custom code. I could find lots of blogs and tutorials about showing angular form errors but not so far a library function we can just pull from NPM.